pkg/pillar/hypervisor/pci.go

```diff
@@ -0,0 +1,200 @@
+// Copyright (c) 2023 Zededa, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+package hypervisor
+
+import (
+	"fmt"
+	"os"
+	"strconv"
+	"strings"
+
+	"github.com/sirupsen/logrus"
+
+	"github.com/lf-edge/eve/pkg/pillar/types"
+)
+
+// define go constants for the flags as defined in include/linux/pci_ids.h
+const (
+	IORESOURCE_BITS      = 0x000000ff
+	IORESOURCE_TYPE_BITS = 0x00001f00
+	IORESOURCE_IO        = 0x00000100
+	IORESOURCE_MEM       = 0x00000200
+	IORESOURCE_REG       = 0x00000300
+	IORESOURCE_IRQ       = 0x00000400
+	IORESOURCE_DMA       = 0x00000800
+	IORESOURCE_BUS       = 0x00001000
+	IORESOURCE_MEM_64    = 0x00100000
+)
+
+type pciResource struct {
+	start uint64
+	end   uint64
+	flags uint64
+	index int
+}
+
+// implement some useful functions on pciResource
+func (r pciResource) size() uint64 {
+	return r.end - r.start + 1
+}
+
+// returns true if the resource is valid
+func (r pciResource) valid() bool {
+	return r.flags != 0 && r.start != 0 && r.end != 0
+}
+
+// returns true if the resource is MEM
+func (r pciResource) isMem() bool {
+	return r.flags&IORESOURCE_TYPE_BITS == IORESOURCE_MEM
+}
+
+func addNoDuplicatePCI(list []typeAndPCI, tap typeAndPCI) []typeAndPCI {
+
+	for _, t := range list {
+		if t.pciLong == tap.pciLong {
+			return list
+		}
+	}
+	return append(list, tap)
+}
+
+type typeAndPCI struct {
+	pciLong string
+	ioType  types.IoType
+}
+
+// check if the PCI device is a VGA device
+// check availability of the VGA file in the sysfs filesystem
+func (t typeAndPCI) isVGA() bool {
+	bootVgaFile := sysfsPciDevices + t.pciLong + "/boot_vga"
+	if _, err := os.Stat(bootVgaFile); err == nil {
+		return true
+	}
+	return false
```

Comment:
```suggestion
    return fileutils.FileExists(nil, bootVgaFile)
```
and import `fileutils "github.com/lf-edge/eve/pkg/pillar/utils/file"`

Reply:
fixed